manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/api/internal/ShardAPI.java

```diff
@@ -0,0 +1,162 @@
+            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
+            @APIResponse(description = "Unauthorized.", responseCode = "401"),
+            @APIResponse(description = "Forbidden.", responseCode = "403"),
+            @APIResponse(description = "Internal error.", responseCode = "500", content = @Content(mediaType = MediaType.APPLICATION_JSON))
+    })
+    @Operation(hidden = true, summary = "Update a Bridge status.", description = "Update a Bridge status.")
+    @PUT
+    public Response updateBridgeStatus() {
+        return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), "Not implemented yet.").build();
+    }
+
+    @APIResponses(value = {
+            @APIResponse(description = "Success.", responseCode = "200",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(type = SchemaType.ARRAY, implementation = ProcessorDTO.class))),
+            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
+            @APIResponse(description = "Unauthorized.", responseCode = "401"),
+            @APIResponse(description = "Forbidden.", responseCode = "403"),
+            @APIResponse(description = "Internal error.", responseCode = "500", content = @Content(mediaType = MediaType.APPLICATION_JSON))
+    })
+    @Operation(hidden = true, summary = "Get Processors to be processed by a shard.", description = "Get Processors to be processed by a shard.")
+    @GET
+    @Path("processors")
+    public Response getProcessors() {
+        String shardId = identityResolver.resolve(jwt);
+        failIfNotAuthorized(shardId);
+        LOGGER.info("Request from Shard for Processors to deploy or delete.");
+        List<Processor> processors = processorService.findByShardIdToDeployOrDelete(shardId);
+        LOGGER.info("Found {} processor(s) to deploy or delete", processors.size());
+        return Response.ok(processors
+                .stream()
+                .map(processorService::toDTO)
+                .collect(toList()))
+                .build();
+    }
+
+    @APIResponses(value = {
+            @APIResponse(description = "Success.", responseCode = "200"),
+            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
+            @APIResponse(description = "Unauthorized.", responseCode = "401"),
+            @APIResponse(description = "Forbidden.", responseCode = "403"),
+            @APIResponse(description = "Internal error.", responseCode = "500", content = @Content(mediaType = MediaType.APPLICATION_JSON))
+    })
+    @Operation(hidden = true, summary = "Update a Processor status.", description = "Update a Processor status.")
+    @PUT
+    @Path("processors")
+    public Response updateProcessorStatus(ProcessorStatusDTO statusDTO) {
```

Comment:
@manstis I'm wondering if this should accept a `List` of `ProcessorStatusDTO`. That way the `Shard` could batch update if it wants to (or just send a single update). Gives us a bit more flexibility in how we handle the updates from the Data plane --> Control Plane and also gives us a "balanced" API e.g:

- The data plane receives a list of Processors to deploy
- The data plane sends a list of Status updates for the control plane to receive

Reply:
@robpblake I know this is being discussed.. I really don't mind.

The change to the `manager` is trivial. There are pros and cons to both.

If we have the `StatusSynchroniser` type class scheduled method thing a batch seems better. 

Singular fitted better when being emitted by the `Controller`.

Reply:
Updated.